django/contrib/admin/options.py

```diff
@@ -1058,7 +1058,14 @@ def response_add(self, request, obj, post_url_continue=None):
         opts = obj._meta
         pk_value = obj._get_pk_val()
         preserved_filters = self.get_preserved_filters(request)
-        msg_dict = {'name': force_text(opts.verbose_name), 'obj': force_text(obj)}
+        obj_url = reverse('admin:%s_%s_change' %
+                          (opts.app_label, opts.model_name),
+                          args=(quote(pk_value),),
+                          current_app=self.admin_site.name)
+        msg_dict = {
+            'name': force_text(opts.verbose_name),
+            'obj': force_text('<a href="{0}">{1}</a>').format(urlquote(obj_url), escape(obj))
```

Comment:
Won't replacing `obj` in the message mean that when *Save and continue editing* is chosen the message will say something like `"saved <a href="...">obj</a>, you may continue editing it below"` where the `<a href>` is a link to the same page you're already on?